src/pynxtools_raman/witec/txt/witec_reader.py

```diff
@@ -0,0 +1,415 @@
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Parser for reading Raman data from
+Witec spectrometers, to be passed to
+Raman nxdl (NeXus Definition Language) template.
+"""
+
+import copy
+import re
+import warnings
+from pathlib import Path
+from typing import Any, Dict, List, Tuple, Union
+
+import numpy as np
+import xarray as xr
+from igor2 import binarywave
+
+from pynxtools_raman.reader_utils import (
+    RamanMapper,
+    _check_valid_value,
+    _re_map_single_value,
+    construct_data_key,
+    construct_entry_name,
+)
+from pynxtools_raman.witec.txt.witec_data_model import WitecHeader, WitecRegion
+from pynxtools_raman.witec.txt.witec_mappings import (
+    UNITS,
+    VALUE_MAP,
+    _construct_date_time,
+    _get_key_value_pair,
+)
+from pynxtools_raman.value_mappers import convert_units, get_units_for_key
+
+
+class MapperWitec(RamanMapper):
```

Comment:
I suggest you try to get rid of this intermediate mapping class if you don't need it. I am only using it in xps because I have many different versions of the parser and I need a uniform interface to them. But again, there's probably many different (better) ways to achieve this.